### PR TITLE
[CARBONDATA-1338] deserialized tableInfo at executor side

### PR DESCRIPTION
**Pull request by cenyuhai**
deserialized TableInfo at executor side

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by cenyuhai** [comment]
ok, to test

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3250/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/655/

**Comment by cenyuhai** [comment]
@chenliang613  @QiangCai

**Comment by jackylk** [comment]
To avoid passing big object to executor, now CarbonScanRDD will serialize TableInfo into serializedTableInfo and send to executor. The tableInfo val in the constructor is only used in driver.

**Comment by jackylk** [comment]
To avoid passing big object to executor, now CarbonScanRDD will serialize TableInfo into serializedTableInfo and send to executor. The tableInfo val in the constructor is only used in driver.

**Comment by cenyuhai** [comment]
can you merge this pr？

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala [hunk 1]
```diff
@@ -254,7 +254,7 @@ class CarbonScanRDD(
 
   private def prepareInputFormatForExecutor(conf: Configuration): CarbonTableInputFormat[Object] = {
     CarbonTableInputFormat.setCarbonReadSupport(conf, readSupport)
-    CarbonTableInputFormat.setTableInfo(conf, tableInfo)
+    CarbonTableInputFormat.setTableInfo(conf, TableInfo.deserialize(serializedTableInfo))
```
Comment (jackylk) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala]: use `getTableInfo`

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3267/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/672/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/673/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3268/

**Comment by jackylk** [comment]
retest this please

**Comment by jackylk** [comment]
retest this please

**Comment by chenliang613** [comment]
add to whitelist

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/674/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3269/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/675/

**Comment by jackylk** [comment]
LGTM

**Comment by jackylk** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3270/

**Pull request closed without merging**